src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -29,58 +29,110 @@ namespace Microsoft.Sarif.Viewer.ErrorList
     public class ErrorListService
     {
         public static readonly ErrorListService Instance = new ErrorListService();
+        private const string VersionRegexPattern = @"\""version\"":\s*\""(?<version>[\d.]+)\""";
 
         public static void ProcessLogFile(string filePath, Solution solution, string toolFormat = ToolFormat.None)
         {
             SarifLog log = null;
 
             string logText;
             string outputPath = null;
+            bool saveOutputFile = true;
 
             if (toolFormat.MatchesToolFormat(ToolFormat.None))
             {
                 logText = File.ReadAllText(filePath);
-                string pattern = @"""version""\s*:\s*""1.0.0""";
-                Match match = Regex.Match(logText, pattern, RegexOptions.Compiled | RegexOptions.Multiline);
+                Match match;
+                
+                // Get the schema version of the unmodified input log
+                using (StringReader reader = new StringReader(logText))
```

Comment:
>StringReader [](start = 23, length = 12)

`var` is ok when type is explicit on the rhs.

Reply:
But wait. You already slurped the whole file. You don't need a reader, just:
```
const int HeadLength = 200;
string head = logText.Substring(0, HeadLength);
Match match = Regex.Match(head, ...);
if (match.Success)
{
    ...
}
```

---
In reply to: [294052626](https://github.com/microsoft/sarif-visualstudio-extension/pull/88#discussion_r294052626) [](ancestors = 294052626)